### PR TITLE
added script for submarine pipeline and jobs

### PR DESCRIPTION
**Pull request by mbiarnes**
@mswiderski 
creates a four jobs 
    pipeline job (for all submarine reps)
    sumbarine-bom
    submarine-runtimes
    submarine-examples

you can see the jobs at
[sumbarine pipeline](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/submarine/job/submarine-pipeline-master/)
[submarine-bom](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/submarine/job/submarine-bom-master/)
[submarine-runtimes](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/submarine/job/submarine-runtimes-master/)
[submarine-examples](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/submarine/job/submarine-examples-master/)

### CODE REVIEW

File: job-dsls/jobs/submarine.groovy [hunk 1]
```diff
@@ -0,0 +1,291 @@
+import org.kie.jenkins.jobdsl.Constants
+
+def javadk=Constants.JDK_VERSION
+def javaToolEnv="KIE_JDK1_8"
+def mvnToolEnv="KIE_MAVEN_3_5_4"
+def mvnVersion="kie-maven-3.5.4"
```
Comment (mswiderski) [inline job-dsls/jobs/submarine.groovy]: would it be possible to get maven 3.6.0 as it is needed for quarkus?

Comment (mbiarnes) [reply job-dsls/jobs/submarine.groovy]: I don't think we have this yet on our machines . have to speak to people

Comment (mswiderski) [reply job-dsls/jobs/submarine.groovy]: ok, let's make this in and then fix the version of maven .. this version of maven is only needed for quarks based examples so we can leave without them for now.

Comment (mareknovotny) [reply job-dsls/jobs/submarine.groovy]: but I think @mbiarnes already prepared everything for version 3.6.0. so we don't need to wait.

**Pull request merged**